src/App/Components/Common/NavProfile/Profile.js

```diff
@@ -0,0 +1,126 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import actions from '../../../../Redux/Actions/notifications';
+import defaultData from '../../../../configs/urls';
+import Triangle from '../Triangle';
+
+const { show, hide } = actions;
+
+export class Profile extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isShown: false,
+      profile: {},
+      notifications: [],
+      notificationsBubble: ''
+    };
+  }
+
+  componentWillReceiveProps({
+    isShown, profile, configs, data
+  }) {
+    this.setState({ isShown });
+    if (!_.isEmpty({ ...profile, ...configs })) this.setState({ profile });
```

Comment:
why this if condition?

Reply:
This is to make sure that the component which will fetch these props have already fetched them and they are available in the store